tests/core/test_pipeline.py

```diff
@@ -188,6 +188,26 @@ def test_pipeline_slice():
     assert 'char' in res_df.columns
 
 
+def test_pipeline_slice_by_label():
+    """Testing something."""
+    drop_num1 = SilentDropStage('num1', label='dropNum1')
+    drop_num2 = SilentDropStage('num2', label='dropNum2')
+    drop_char = SilentDropStage('char', label='dropChar')
+    pipeline = PdPipeline([drop_num1, drop_num2, drop_char])
+    assert len(pipeline) == 3
+    pipeline = pipeline[['dropNum1', 'dropNum2']]
```

Comment:
the way it's implemented now, (I think) one can pass as labels integers, and slice here using integers which would make it a bit confusing I think. I would force the labels to be strings, and ensure people cannot pass a list of integers to slice. WDYT?

Reply:
Notice that we are not going to disable int base slicing or access, as pipelines are first-and-formost lists of pipeline stages, and they should easily slice-able / index-accessible with integers. It's in the documentation.